File: CIS22B_FinalProject_Workflow/testWorkflow/Book.vb
Public Class Book
    Private Property title As String
        Get
            Return Nothing
        End Get
        Set(value As String)
        End Set
    End Property

    Private Property isbn As String
        Get
            Return Nothing
        End Get
        Set(value As String)
        End Set
    End Property

    Private Property author As String
        Get
            Return Nothing
        End Get
        Set(value As String)
        End Set
    End Property

    Private Property publisher As String
        Get
            Return Nothing
        End Get
        Set(value As String)
        End Set
    End Property

    Private Property wholesaleCost As Double
        Get
            Return Nothing
        End Get
        Set(value As Double)
        End Set
    End Property

    Private Property retailPrice As Double
        Get
            Return Nothing
        End Get
        Set(value As Double)
        End Set
    End Property

    ''' <summary>
    ''' MM/DD/YYYY
    ''' </summary>
    Private Property dateAdded As String
        Get
            Return Nothing
        End Get
        Set(value As String)
        End Set
    End Property

    ''' <summary>
    ''' a nuber that is unique to each specific book
    ''' </summary>
    Private Property bookIdentifier As Int
        Get
            Return Nothing
        End Get
        Set(value As Int)
        End Set
    End Property

    Public Function getAuthor() As String
    End Function

    Public Function setAuthor(auth As String) As Void
    End Function

    Public Function getIsbn() As String
    End Function

    Public Sub setIsbn(isbn As String)

    End Sub

    Public Function getTitle() As String
    End Function

    Public Sub setTitle(nam As String)

    End Sub

    Public Function getPublisher() As String
    End Function

    Public Sub setPublisher(pub As String)

    End Sub

    Public Function getRetailPrice() As Double
    End Function

    Public Sub setRetailPrice(price As Double)

    End Sub

    Public Function getWholesaleCost() As Double
    End Function

    Public Sub setWholesaleCost(cost As Double)

    End Sub

    ''' <summary>
    ''' default constructor
    ''' </summary>
    Public Function Book() As testWorkflow.Book
    End Function

    Public Function Book(title As String, author As String, isbn As String, publisher As String, wholesaleCost As Double, retailPrice As String, dateAdded As String, identifier As String) As testWorkflow.Book
    End Function

    Public Function getDateAdded() As String
    End Function

    Public Sub setDateAdded(added As String)

    End Sub

    Public Function getIdentifier() As Int
    End Function

    Public Sub setIdentifier(identifier As Int)

    End Sub
End Class

File: CIS22B_FinalProject/CIS22B_FinalProject/Book.cpp
#define _CRT_SECURE_NO_WARNINGS
#include "Book.h"

Book::Book()
{
	author = "default author";
	bookIdentifier = -1;
	dateAdded = "XX/XX/XXXX";
	isbn = "defualt isbn";
	publisher = "default publisher";
	retailPrice = 0.0;
	title = "default title";
	wholesaleCost = 0.0;
}

Book::Book(string titl, string auth, string Isbn, string publish, double wholesaleCos, double retailPric, string dateAdd, int identifier)
{
	title = titl;
	author = auth;
	isbn = Isbn;
	publisher = publish;
	wholesaleCost = wholesaleCos;
	retailPrice = retailPric;
	dateAdded = dateAdd;
	bookIdentifier = identifier;
}

void Book::setAll(string titl, string auth, string Isbn, string publish, double wholesaleCos, double retailPric, string dateAdd, int identifier)
{
	title = titl;
	author = auth;
	isbn = Isbn;
	publisher = publish;
	wholesaleCost = wholesaleCos;
	retailPrice = retailPric;
	dateAdded = dateAdd;
	bookIdentifier = identifier;
}

const string Book::getAuthor()
{
	return author;
}

const string Book::getDateAdded()
{
	return dateAdded;
}

const int Book::getIdentifier()
{
	return bookIdentifier;
}

const string Book::getIsbn()
{
	return isbn;
}

const string Book::getPublisher()
{
	return publisher;
}

const double Book::getRetailPrice()
{
	return retailPrice;
}

const string Book::getTitle()
{
	return title;
}

const double Book::getWholesaleCost()
{
	return wholesaleCost;
}

void Book::setAuthor(string auth)
{
	author = auth;
}

void Book::setDateAdded(string dateAdd)
{
	dateAdded = dateAdd;
}

void Book::setIdentifier(int identifier)
{
	bookIdentifier = identifier;
}

void Book::setIsbn(string Isbn)
{
	isbn = Isbn;
}

void Book::setPublisher(string publish)
{
	publisher = publish;
}

void Book::setRetailPrice(double retailPric)
{
	retailPrice = retailPric;
}

void Book::setTitle(string titl)
{
	title = titl;
}

void Book::setWholesaleCost(double wholesaleCos)
{
	wholesaleCost = wholesaleCos;
}

ostream& operator<<(ostream& out, Book& book)
{
	out << fixed << setprecision(2);			// makes doubles round to two digits
	out << "ISBN: " << book.getIsbn() << endl;
	out << "Title: " << book.getTitle() << endl;
	out << "Author: " << book.getAuthor() << endl;
	out << "Publisher: " << book.getPublisher() << endl;
	out << "Date Added: " << book.getDateAdded() << endl;
	out << "Wholesale Cost: $" << book.getWholesaleCost() << endl;
	out << "Retail Price: $" << book.getRetailPrice() << endl;
	return out;
}